src/vs/workbench/contrib/replNotebook/browser/repl.contribution.ts

```diff
@@ -129,17 +133,34 @@ export class ReplDocumentContribution extends Disposable implements IWorkbenchCo
 					const scratchpad = this.configurationService.getValue<boolean>(NotebookSetting.InteractiveWindowPromptToSave) !== true;
 					const ref = await this.notebookEditorModelResolverService.resolve({ untitledResource: resource }, 'jupyter-notebook', { scratchpad, viewType: 'repl' });
 
+					const notebookUri = ref.object.notebook.uri;
```

Comment:
from what I read, we are not caching the editor inputs anywhere else, e.g. text editor, custom editor, notebook. I wonder if this is any misunderstanding of how it should be cached.

Reply:
text editors do cache inputs, but they might currently be the only one